packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,390 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnGatewayRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualGateway, VirtualGateway } from './virtual-gateway';
+import { IVirtualService } from './virtual-service';
+
+/**
+ * Interface for which all GatewayRoute based classes MUST implement
+ */
+export interface IGatewayRoute extends cdk.IResource {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteName: string,
+
+  /**
+   * The Amazon Resource Name (ARN) for the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteArn: string;
+
+  /**
+   * The VirtualGateway the GatewayRoute belongs to
+   *
+   * @attribute
+   */
+  readonly virtualGateway: IVirtualGateway;
+}
+
+/**
+ * Basic configuration properties for a GatewayRoute
+ */
+export interface GatewayRouteBaseProps {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @default - an automatically generated name
+   */
+  readonly gatewayRouteName?: string;
+
+  /**
+   * What protocol the route uses
+   */
+  readonly routeSpec: GatewayRouteSpec;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface HttpGatewayRouteMatch {
```

Comment:
I wonder - can we move all `RouteMatch` and `RouteSpec`-related types to a separate file? Might make it a little easier to read.

Reply:
Done!